File: backend/src/main/java/io/blog/devlog/domain/mail/controller/EmailController.java
package io.blog.devlog.domain.mail.controller;

import io.blog.devlog.domain.mail.dto.AuthenticationMessage;
import io.blog.devlog.domain.mail.service.EmailService;
import io.blog.devlog.domain.mail.types.MessageType;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;

@RestController
@RequiredArgsConstructor
public class EmailController {

    private final EmailService emailService;

    /* TEST CODE */
    @PostMapping("/send")
    void sendEmail(@RequestBody AuthenticationMessage message) {
        HashMap<String, String> params = new HashMap<>();
        params.put("authCode", message.getAuthCode());
        emailService.sendEmail(MessageType.EMAIL, message.getEmail(), message.getSubject(), params);
    }
}

File: backend/mail-service/src/main/kotlin/io/blog/mail/mail/controller/EmailController.kt
package io.blog.mail.mail.controller

import io.blog.mail.mail.message.AuthenticationMessage
import io.blog.mail.mail.service.EmailService
import io.blog.mail.mail.types.MessageType
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class EmailController {
    @Autowired
    private lateinit var emailService : EmailService

    @PostMapping("/send")
    fun sendEmail(@RequestBody message: AuthenticationMessage) {
        val params = HashMap<String, String>()
        params["authCode"] = message.authCode
        emailService.sendEmail(MessageType.EMAIL, message.email, message.subject, params)
    }
}